iocore/net/OCSPStapling.cc

```diff
@@ -145,15 +148,19 @@ ssl_stapling_init_cert(SSL_CTX *ctx, X509 *cert, const char *certname)
     return false;
   }
 
-  cinf = (certinfo *)SSL_CTX_get_ex_data(ctx, ssl_stapling_index);
-  if (cinf) {
+  certinfo_map *map = static_cast<certinfo_map *>(SSL_CTX_get_ex_data(ctx, ssl_stapling_index));
+  if (map && map->find(cert) != map->end()) {
     Note("certificate already initialized for %s", certname);
     return false;
   }
 
-  cinf = (certinfo *)OPENSSL_malloc(sizeof(certinfo));
+  if (!map) {
```

Comment:
Why is this done before the `cinf` checks? I don't see it used there, other than cleanup.

Reply:
We need the map to place the cinf into.  I guess we could do the allocs in the other order, but then we'd need to clean up the cinfo if the creation of the map failed.